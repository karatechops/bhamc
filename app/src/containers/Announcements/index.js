import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import Spinning from 'grommet/components/icons/Spinning';
import Announcement from 'components/Announcement';
import * as AnnouncementsAcctions from './actions';

export class Announcements extends Component {
  componentDidMount() {
    this.props.dispatch(AnnouncementsAcctions.loadData());
  }

  render() {
    const { data, error, request } = this.props;

    const posts = (!request && data.length > 0)
      ? data.map(({ title, paragraph, id }) =>
        <Announcement
          title={title}
          paragraph={paragraph}
          key={`announcement-${id}`}
        />,
      )
      : <Spinning size="medium" />;

    return (
      <div>
        {error || posts}
      </div>
    );
  }
}

function mapStateToProps(state) {
  const { request, error, data } = state.announcements;
  return {
    request, error, data,
  };
}

Announcements.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      paragraph: PropTypes.string,
    }),
  ),
  dispatch: PropTypes.func,
  error: PropTypes.string,
  request: PropTypes.bool,
};

export default connect(mapStateToProps)(Announcements);
