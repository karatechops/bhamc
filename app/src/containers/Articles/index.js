import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import Box from 'grommet/components/Box';
import WithLoading from 'components/WithLoading';
import { Article } from 'components';
import * as ArticlesActions from './actions';

export class Articles extends Component {
  componentDidMount() {
    const { query } = this.props;
    const id = query && query.id;
    const category = query && query.category;
    this.props.dispatch(ArticlesActions.loadData(id, category));
  }

  render() {
    const { data, error, request } = this.props;

    const post = (!request && data && data.hasOwnProperty('html'))
      ? <Article __html={data.html} title={data.title} />
      : undefined;

    return (
      <Box align="center" full="horizontal">
        <WithLoading request={request}>
          {error || post}
        </WithLoading>
      </Box>
    );
  }
}

function mapStateToProps(state) {
  const { request, error, data } = state.articles;
  return {
    request, error, data,
  };
}

Articles.propTypes = {
  data: PropTypes.shape({
    type: PropTypes.string,
    title: PropTypes.string,
    html: PropTypes.string,
  }),
  dispatch: PropTypes.func,
  error: PropTypes.string,
  request: PropTypes.bool,
  query: PropTypes.shape({
    id: PropTypes.string,
    category: PropTypes.string,
  }),
};

export default connect(mapStateToProps)(Articles);
