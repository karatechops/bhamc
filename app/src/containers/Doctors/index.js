import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import Box from 'grommet/components/Box';
import DoctorListItem from 'components/DoctorListItem';
import WithLoading from 'components/WithLoading';
import * as DoctorsActions from './actions';

export class Doctors extends Component {
  componentDidMount() {
    this.props.dispatch(DoctorsActions.loadData());
  }

  render() {
    const { data, error, request } = this.props;

    const posts = (!request && data.length > 0)
      ? data.map(({ biography, name, profile, id }) =>
        <DoctorListItem
          biography={biography}
          name={name}
          profile={profile}
          id={id}
          key={`doctor-${id}`}
        />,
      )
      : undefined;

    return (
      <Box align="center">
        <WithLoading request={request}>
          {error ||
            <Box
              wrap
              full="horizontal"
              direction="row"
            >
              {posts}
            </Box>
        }
        </WithLoading>
      </Box>
    );
  }
}

function mapStateToProps(state) {
  const { request, error, data } = state.doctors;
  return {
    request, error, data,
  };
}

Doctors.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      paragraph: PropTypes.string,
    }),
  ),
  dispatch: PropTypes.func,
  error: PropTypes.string,
  request: PropTypes.bool,
};

export default connect(mapStateToProps)(Doctors);
