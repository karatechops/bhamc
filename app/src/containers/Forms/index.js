import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import Box from 'grommet/components/Box';
import { DownloadLink } from 'components';
// import WithLoading from 'components/WithLoading';
// import * as FormsActions from './actions';

export class Forms extends Component {
  componentDidMount() {
    // this.props.dispatch(FormsActions.loadData());
  }

  render() {
    // const { data, error, request } = this.props;

    /* const posts = (!request && data.length > 0)
      ? data.map(({ biography, name, profile, id }) =>
        <DoctorListItem
          biography={biography}
          name={name}
          profile={profile}
          key={`doctor-${id}`}
        />,
      )
      : undefined; */

    return (
      <Box align="center" pad={{ between: 'medium' }}>
        {/* <WithLoading request={request}>
          {error || posts}
        </WithLoading> */}
        <DownloadLink
          label="Anesthesia and Surgical Consent Form"
          href="http://bhamc.com/docs/Anesth%20and%20Surgical%20Consent%20Form.pdf"
        />
        <DownloadLink
          label="Drop-Off Consent Form"
          href="http://bhamc.com/docs/Drop-Off%20Consent%20Form.pdf"
        />
        <DownloadLink
          label="Boarding Consent Form"
          href="http://bhamc.com/docs/BoardingDropOffForm.pdf"
        />
      </Box>
    );
  }
}

function mapStateToProps(state) {
  const { request, error, data } = state.doctors;
  return {
    request, error, data,
  };
}

Forms.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      paragraph: PropTypes.string,
    }),
  ),
  dispatch: PropTypes.func,
  error: PropTypes.string,
  request: PropTypes.bool,
};

export default connect(mapStateToProps)(Forms);
