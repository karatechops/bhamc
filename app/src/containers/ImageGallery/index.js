import React, { Component, PropTypes } from 'react';
import Box from 'grommet/components/Box';
import Button from 'grommet/components/Button';
import Image from 'grommet/components/Image';
import Paragraph from 'grommet/components/Paragraph';
import { ImageThumb } from 'components';

export class ImageGallery extends Component {
  constructor(props) {
    super(props);
    this.state = {
      activeIndex: 0,
    };
    this.onImageClick = this.onImageClick.bind(this);
  }
  onImageClick(activeIndex) {
    this.setState({ activeIndex });
  }
  render() {
    const { images } = this.props;
    const { activeIndex } = this.state;
    const imageThumbs = (images && images.length > 0)
      ? images.map((image, imageIndex) =>
        <Button
          onClick={this.onImageClick.bind(this, imageIndex)}
          key={`gallery-thumb-${imageIndex}`}
        >
          <ImageThumb
            path={image.original}
          />
        </Button>)
      : undefined;
    return (
      <Box full="horizontal">
        <Box
          full="horizontal"
          colorIndex="light-2"
          pad="medium"
          align="center"
        >
          <Image src={images[activeIndex].original} />
        </Box>
        { images[activeIndex].description &&
          <Box
            align="center"
            colorIndex="light-2"
            pad={{
              horizontal: 'small',
            }}
            textAlign="center"
          >
            <Paragraph size="large" margin="small">
              {images[activeIndex].description}
            </Paragraph>
          </Box>
        }
        { imageThumbs &&
          <Box
            pad={{
              vertical: 'medium',
            }}
            direction="row"
            justify="center"
            responsive={false}
            wrap
          >
            {imageThumbs}
          </Box>
        }
      </Box>
    );
  }
}

ImageGallery.propTypes = {
  images: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string,
      description: PropTypes.string,
      original: PropTypes.string,
      thumb: PropTypes.string,
      icon: PropTypes.string,
    }),
  ),
};

export default ImageGallery;