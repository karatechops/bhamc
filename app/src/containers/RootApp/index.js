/* @flow */
import React, { Component, PropTypes } from 'react';
import GrommetApp from 'grommet/components/App';
import { AppContainer as ReactHotLoader } from 'react-hot-loader';
import { Nav, PageMarquee, Footer } from 'components';

// eslint-disable-next-line react/prefer-stateless-function
class RootApp extends Component {
  render() {
    return (
      <ReactHotLoader>
        <GrommetApp centered={false}>
          <Nav />
          <PageMarquee />
          {this.props.children}
          <Footer />
        </GrommetApp>
      </ReactHotLoader>
    );
  }
}

RootApp.propTypes = {
  children: PropTypes.node.isRequired,
};

export default RootApp;
