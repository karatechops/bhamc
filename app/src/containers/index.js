/* GENERATOR: Assemble all containers for export */
export RootApp from './RootApp';
export Announcements from './Announcements';
