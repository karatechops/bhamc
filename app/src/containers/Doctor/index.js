import React, { Component, PropTypes } from 'react';
import { connect } from 'react-redux';
import Box from 'grommet/components/Box';
import DoctorPost from 'components/DoctorPost';
import WithLoading from 'components/WithLoading';
import * as DoctorsActions from '../Doctors/actions';

export class Doctor extends Component {
  componentDidMount() {
    this.props.dispatch(DoctorsActions.loadData());
  }
  onImageClick(imageIndex) {
    console.log(imageIndex)
  }
  findDoctor(id, doctors) {
    return doctors.find(doctor => doctor.id.toString() === id);
  }
  render() {
    const { data, error, request, id } = this.props;
    const doctor = (!request && data.length > 0)
      ? this.findDoctor(id, data)
      : undefined;
    const post = (doctor)
      ? <DoctorPost
        biography={doctor.biography}
        name={doctor.name}
        profile={doctor.profile}
        images={doctor.images}
        onImageClick={this.onImageClick}
      />
      : undefined;

    return (
      <Box align="center">
        <WithLoading request={request}>
          {error || post }
        </WithLoading>
      </Box>
    );
  }
}

function mapStateToProps(state) {
  const { request, error, data } = state.doctors;
  return {
    request, error, data,
  };
}

Doctor.propTypes = {
  data: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number,
      title: PropTypes.string,
      paragraph: PropTypes.string,
    }),
  ),
  id: PropTypes.string.isRequired,
  dispatch: PropTypes.func,
  error: PropTypes.string,
  request: PropTypes.bool,
};

export default connect(mapStateToProps)(Doctor);
