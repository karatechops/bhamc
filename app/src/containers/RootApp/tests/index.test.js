import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';
import { HomeApp } from 'containers';
import RootApp from '../index';


describe('<RootApp />', () => {
  it('should render with children', () => {
    const wrapper = shallow(
      <RootApp>
        ðŸ‘¶ðŸ‘¶ðŸ‘¶ðŸ‘¶
      </RootApp>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with home app children', () => {
    const wrapper = shallow(
      <RootApp>
        <HomeApp>
          ðŸ‘¶ðŸ‘¶ðŸ‘¶ðŸ‘¶
        </HomeApp>
      </RootApp>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
