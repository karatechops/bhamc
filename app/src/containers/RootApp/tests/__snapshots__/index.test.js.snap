exports[`<RootApp /> should render with children 1`] = `
<AppContainer>
  <App
    centered={false}>
    ğŸ‘¶ğŸ‘¶ğŸ‘¶ğŸ‘¶
    <Connect(ErrorHandler) />
    <Connect(UserNotifications) />
  </App>
</AppContainer>
`;

exports[`<RootApp /> should render with home app children 1`] = `
<AppContainer>
  <App
    centered={false}>
    <HomeApp>
      ğŸ‘¶ğŸ‘¶ğŸ‘¶ğŸ‘¶
    </HomeApp>
    <Connect(ErrorHandler) />
    <Connect(UserNotifications) />
  </App>
</AppContainer>
`;
